src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -930,7 +930,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     var lazySemanticModel = body == null ? null : new Lazy<SemanticModel>(() =>
                     {
                         var syntax = body.Syntax;
-                        var semanticModel = (CSharpSemanticModel)_compilation.GetSemanticModel(syntax.SyntaxTree);
+                        var semanticModel = (CSharpSemanticModel)AnalyzerDriver.GetOrCreateCachedSemanticModel(syntax.SyntaxTree, _compilation, _cancellationToken);
```

Comment:
> However, with the new compilation events based IDE analyzer driver, the IDE already has a semantic model which needs to be used across diagnostic computations for different analyzers. This is the semantic model on which CompilationWithAnalyzers explicitly invokes GetDiagnostics to force binding symbols in the document, and passes down to the analyzers.

I don't understand why it is important to use the same semantic model. Can you step me through the scenario?